component/main.jsonnet

```diff
@@ -18,6 +18,36 @@ local namespace = kube.Namespace(params.namespace) {
   },
 };
 
+local namespacedName(name, namespace='') = {
+  local namespacedName = std.splitLimit(name, '/', 1),
+  local ns = if namespace != '' then namespace else params.namespace,
+  namespace: if std.length(namespacedName) > 1 then namespacedName[0] else ns,
+  name: if std.length(namespacedName) > 1 then namespacedName[1] else namespacedName[0],
+};
+
+local vpa_resources() = [
+  local vpa = std.get(params.autoscaler, name);
+  {
+    apiVersion: 'autoscaling.k8s.io/v1',
+    kind: 'VerticalPodAutoscaler',
+    metadata: {
+      name: namespacedName(name, '').name,
+      namespace: namespacedName(name, '').namespace,
+    },
+    spec: {
+      targetRef: {
+        apiVersion: 'apps/v1',
+        kind: std.get(vpa, 'kind', 'Deployment'),
+        name: namespacedName(name, '').name,
+      },
+      updatePolicy: {
+        updateMode: std.get(vpa, 'mode', 'Off'),
+      },
+    },
+  }
```

Comment:
For something like this I would recommend that you expose the complete spec in the config hierarchy.  Something like:

```
autoscaler:
  foo/bar:
    kind: StatefulSet
    mode: Auto
    spec: # can contain arbitrary autoscaler spec
      resourcePolicy:
        containerPolicies:
          - containerName: '*'
            minAllowed:
              cpu: 100m
              memory: 50Mi
            maxAllowed:
              cpu: 1
              memory: 500Mi
            controlledResources: ["cpu", "memory"]     
```

That way you have a simple config for most cases, but for future special cases you won't need to update the component all the time.

Reply:
done